### DISCUSSION TITLE
BTC Prague

### DISCUSSION THREAD

**Discussion by Kixunil** [rust-bitcoin/rust-bitcoin#1535] [General]
I was invited to attend [BTC Prague](https://www.btcprague.com/) which will have a separate tech event (Core & co, LN, nodes...). I would like to do some promotion for rust-bitcoin because a) I didn't do this yet b) I feel this library/ecosystem is a bit under-promoted (except for LN) :)

My idea so far is to make a workshop with this agenda:

* Why Rust? - quick intro, the main advantages...
* Overview of rust-bitcoin ecosystem - this crate but others as well, how they are related, what they provide...
* Hands-on: let's make a nano wallet - we will make a trivial Taproot-only CLI wallet that can receive and send BTC and we will test it on regtest
    - `init` command: creates a wallet file
    - `address` command: returns an unused address
    - `balance` command: scans the chain for incoming transactions
    - `send` command: sends a specified amount of coins to an address

Any ideas what to improve? Or perhaps a different project that better demonstrates the capabilities of the ecosystem? (should be doable within ~1h I think)

If you're attending too we can meet and perhaps do it together or something. (Or just chat and hack together!) If you're considering buying a ticket, there's a good chance some of you may get invited as well because of your participation in various Bitcoin projects, so perhaps wait a bit. The organizers told me they want to invite numerous technical people from various projects.

**Comment by apoelstra** [comment]
Very cool! I cannot attend in person but this sounds like a great idea. Though even what you've listed might take more than an hour, I'd guess.

**Comment by sanket1729** [comment]
Looks exciting. Unfortunately, I cannot be there in Prague too. I heard good things about the event last year from Ruben, so I am sure it would be a great platform for rust-bitcoin. 

It looks too much for one hour if you plan for others to do live coding. I presume you would already have lots of skeleton code and would fill in small parts(5-10 LOC) of sign API/ address API etc. Would be cool if you can demonstrate script spend and a key spend too! Perhaps, as an assignment you can also tease about multi-party taproot transactions with taproot Psbts. 


I demonstrated spending a taproot descriptor last year advancing bitcoin in London. It didn't involve any coding, just using the CLI and understanding miniscript. https://github.com/sanket1729/adv_btc_workshop/blob/master/workshop.md

**Reply by Kixunil** [reply]
> I heard good things about the event last year from Ruben, so I am sure it would be a great platform for rust-bitcoin.

Different organizers but I was at another conference they organized and it was crazy well-organized!

> I presume you would already have lots of skeleton code

I expect I will.

I didn't receive any official time limit yet but that's what I expect it could be. I wouldn't mind extending it if it makes sense (there weill be free-form space so at worst people could move there). We'll see.

> Would be cool if you can demonstrate script spend and a key spend too!

I wish I could! I don't have my hopes high yet.

**Comment by tcharding** [comment]
I'm interested. I just applied for a developer ticket to Miami but for the same money it would cost me to fly to Miami and stay a week I could fly to and stay in Prague for 3 weeks! I'm hoping to come over to Atlanta later in the year anyways so a chance to meet the European freaks would be sweet. I'd be happy to do the tutorial with you @Kixunil if I did come over.

**Comment by tcharding** [comment]
Is it ok if I submit this to tabconf 2023 as well?

**Reply by Kixunil** [reply]
No problem, we need more Rust devs!

**Reply by tcharding** [reply]
Sweet, I'll write it up. Anyone else here who is coming to tabconf and wants to jump on board, holla at me.

**Reply by tcharding** [reply]
https://github.com/TABConf/2023.tabconf.com/issues/34

**Comment by tcharding** [comment]
I've created a repository for workshop material so we can share it and colaborate: https://github.com/rust-bitcoin/workshop
